python/ray/tests/test_basic_3.py

```diff
@@ -184,5 +184,22 @@ def f():
     assert len(ready) == 1000, len(ready)
 
 
+def test_actor_killing(shutdown_only):
+    # This is to test create and kill an actor immediately
+    import ray
+    ray.init(num_cpus=1)
+
+    @ray.remote(num_cpus=1)
+    class Actor:
+        def foo(self):
+            return None
+
+    worker_1 = Actor.remote()
+    ray.kill(worker_1)
```

Comment:
Can you actually test with `ray.kill(worker_1, no_restart=False)` and create an actor with `max_restarts=-1`, and see if this works as expected?

Reply:
With no_restart=False it'll hang. Without, it'll exit as expected.

Reply:
what do you think is the correct behavior here for the setup?

Reply:
Hmm I think actor should be restarted until restart happens `max_restarts ` times. So you can test with 

`no_restart=False` & max_restarts=1

and make sure it is alive the first time, and killed in the second time (then that's the most correct behavior)

Reply:
It'll be alive the first time, but still not being killed the second time :( 
Let me investigate more.